### ISSUE TITLE
Support multisite

### ISSUE THREAD

**Issue by carlalexander** [carlalexander/debops-wordpress#3]
Supporting multisite will probably be a bit more challenging. Will have to test the interaction with the varnish proxy host and other factors.

**Comment by drybjed** [comment]
I thought about WP multisite a bit but after [reading articles against it](https://www.ostraining.com/blog/general/multisites-bad-idea/) I would try to create a role where you can define unix user accounts and create multiple wordpress installations at once. That would probably require defining nginx servers in a separate list as well, but it's still easier than configuring them from scratch.

Alternatively, create a wordpress role which focuses on a unix user account, and allow easy creation of multiple sites by creating a custom playbook with only small number of variables to change if needed.

**Comment by carlalexander** [comment]
I think, in the long run, you should have both the ability to install multiple sites like `nginx servers` in `debops.nginx`. I think some people might want that as well. I have mixed feelings about it, but as long as it's handled like you described, it's safe.

As for Multisite, it's safe. I don't agree with that article at all. The entire wordpress.com is just one giant multisite installation. It's just as safe as any other site. It's just different needs.

In practice, it's the opposite situation that's more unsafe. Unless people take all the necessary precautions like you outlined, you end up with a soup kitchen server which is very common with WordPress shared hosts. Those are breeding ground for infections and quite a bit of work to cleanup.

**Comment by drybjed** [comment]
Yeah, I suppose Multisite has a reason to exists if you specifically want a "network". In that case probably creating separate `wordpress_mu` role would work, depending on how many differences there would be from regular setup.

Creating roles that allow for multiple instances of a webapplication with `debops.nginx`, `debops.php5`, and similar roles is currently problematic, because these roles are desinged for single use from a role dependency. In other words, if you want to create multiple Wordpress sites with separate unix users, you need to pass a list with multiple server configurations to nginx, or play with nginx maps and namespaced variables... which might be a better solution for this, although i haven't tested that yet, just looked like a promising solution.

**Comment by carlalexander** [comment]
Right, that's why I mentioned in the long run! :wink:

I might do a separate `wordpress_mu` role, but I'll have to look into it more. WP-CLI makes it easy to pick between the two so that part is easy. The issue is more with what needs to change with the rest of the stack. That'll require some testing on my part.

**Comment by foxpcteam** [comment]
Hi, great project! when you think multisite support will be ready?

**Comment by carlalexander** [comment]
First priority is HTTPS support then this. I'm slowing down a bit for summer, but I'd like to have this done in August.

**Comment by carlalexander** [comment]
Just pushed the support for multisite. You can't use it to migrate a site to multisite, but it'll setup subdirectory or subdomain network installation.

I'll have to do a pass at the wiki to add a page for it.

**Comment by Ichag** [comment]
Can we do anything to support your work to document the steps to a multisite installation?

**Comment by caramiame** [comment]
[quote]WordPress Multisite is great, but what most people actually want is multiple networks. In this talk (good for all experience levels), I’ll explain what networks are, why you’d use them, and how to unlock this powerful and hidden WordPress core feature.[/quote]

re: http://wordpress.tv/2014/07/26/john-james-jacoby-multisite-and-multi-network/

**Comment by carlalexander** [comment]
I plan on tackling it this week @Ichag. The documentation is roughly just a question of writing what needs to be done in the configuration file to configure a multisite server.

**Comment by carlalexander** [comment]
Alright, you can find the [multisite cookbook](https://github.com/carlalexander/debops-wordpress/wiki/How-to-create-a-multisite-network) in the wiki.

**Comment by caramiame** [comment]
thanks very much! 

```
 On Tuesday, October 6, 2015 3:37 PM, Carl Alexander <notifications@github.com> wrote:
```

 Closed #3.—
